python/ray/_raylet.pyx

```diff
@@ -2059,14 +2059,17 @@ cdef class CoreWorker:
         self.cgname_to_eventloop_dict = {}
         self.fd_to_cgname_dict = {}
 
+        def asyncio_thread_target(event_loop):
+            event_loop.run_forever()
+            # Finish running the existing actor method calls and then exit.
+            remaining = asyncio.Task.all_tasks(event_loop)
+            event_loop.run_until_complete(asyncio.gather(*remaining))
```

Comment:
This changes the current behavior: all running method calls/coroutines will finish (boost fiber thread can be successfully joined as a result). This is consistent with threaded actor and sync actor. Is this something we want? @simon-mo @edoakes

Reply:
actually no. you want to cancel all the running tasks first instead of waiting for them to finish. it is common for an async actor task do 
```python
while True:
    # do something
    await asyncio.sleep(1)
```

Reply:
Isn't it more controlled if user's actor has coordinated exit:

```
async def run_forever_until_stop():
while not stopped:
   # do something
   await asyncio.sleep(1)
# do some clean up

async def stop():
  stopped = true
  ray.actor.exit_actor()
```

Reply:
true but it's a behavior breaking change

Reply:
Updated, PTAL.